src/storage/src/hummock/store/version.rs

```diff
@@ -292,10 +367,22 @@ impl HummockReadVersion {
                     self.staging
                         .imm
                         .retain(|imm| imm.epoch() > max_committed_epoch);
+
+                    self.staging
+                        .merged_imm
+                        .retain(|merged_imm| merged_imm.epoch() > max_committed_epoch);
+
                     self.staging.sst.retain(|sst| {
                         sst.epochs.first().expect("epochs not empty") > &max_committed_epoch
                     });
 
+                    assert!(self
+                        .staging
+                        .imm
+                        .iter()
+                        .chain(self.staging.merged_imm.iter())
+                        .all(|imm| imm.epoch() > max_committed_epoch));
+
```

Comment:
This assertion seems unneccessary since the all condition is exactly the same as the retain condition above.